### PR TITLE
consensus-logic: Remove redundant messaging

### PR DESCRIPTION
**Pull request by bewakes**
## Description
This PR removes the cycle between FCM worker and CSM worker by having CSM send NewState message to FCM via watch channel instead as ForkChoiceMessage::NewState via mpsc channel.

### Type of Change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update
-   [x] Refactor
-   [ ] New or updated tests

## Checklist

-   [x] I have performed a self-review of my code.
-   [x] I have commented my code where necessary.
-   [ ] I have updated the documentation if needed.
-   [ ] My changes do not introduce new warnings.
-   [ ] I have added tests that prove my changes are effective or that my feature works.
-   [x] New and existing tests pass with my changes.

### CODE REVIEW

**Comment by bewakes** [comment]
The conversation from https://github.com/alpenlabs/strata/pull/434#issuecomment-2461419745 can be continued here.

**Review by delbonis — CHANGES_REQUESTED** [review]
Sorry, made this review last week but forgot to actually submit it.

File: crates/consensus-logic/src/csm/worker.rs [hunk 1]
```diff
@@ -164,18 +149,13 @@ fn process_msg<D: Database>(
...
-            if *idx > next_exp_idx {
-                let missed_ev_cnt = idx - next_exp_idx;
-                warn!(%missed_ev_cnt, "applying missed sync events");
-                for ev_idx in next_exp_idx..*idx {
-                    trace!(%ev_idx, "running missed sync event");
-                    handle_sync_event_with_retry(
-                        state, engine, ev_idx, status_rx, fcm_msg_tx, shutdown,
-                    )?;
+            for ev_idx in next_exp_idx..=*idx {
+                if ev_idx < *idx {
```
[the remark this answers is not in the document]
Comment (bewakes) [reply crates/consensus-logic/src/csm/worker.rs:153]: haha, just wanted to make it a bit concise.

File: crates/status/src/status_manager.rs [hunk 2]
```diff
@@ -16,12 +20,82 @@ pub enum StatusError {
...
+        &self.csm
+    }
+
+    pub fn cl(&self) -> &watch::Receiver<ClientState> {
+        &self.cl
+    }
+
+    pub fn l1(&self) -> &watch::Receiver<L1Status> {
+        &self.l1
+    }
```
[the remark this answers is not in the document]
Comment (bewakes) [reply crates/status/src/status_manager.rs:47]: Created a ticket for this
https://alpenlabs.atlassian.net/browse/STR-626

**Review by delbonis — CHANGES_REQUESTED** [review]
Those changes to the status tracker logic you split out as a separate ticket ideally should have been done in this PR, since you're actively working on related code that would benefit from the interface being improved.

File: crates/consensus-logic/src/fork_choice_manager.rs [hunk 3]
```diff
@@ -275,37 +240,66 @@ pub fn tracker_task<D: Database, E: ExecEngineCtl>(
...
         return Err(e);
     }
 
     Ok(())
 }
 
+#[allow(clippy::large_enum_variant)]
+enum FcmIncoming {
+    FcmMsg(Option<ForkChoiceMessage>),
+    StateUpdate(Option<ClientState>),
```
[the remark this answers is not in the document]
Comment (bewakes) [reply crates/consensus-logic/src/fork_choice_manager.rs]: This was done to make the block_on code concise, but with the new `Abort` variant, the options are removed now.

File: crates/consensus-logic/src/fork_choice_manager.rs [hunk 4]
```diff
@@ -444,6 +412,31 @@ fn process_fc_message<D: Database, E: ExecEngineCtl>(
...
+) -> anyhow::Result<()> {
+    let sync = cs.sync().expect("fcm: client state missing sync data");
+
+    let csm_tip = sync.chain_tip_blkid();
+    debug!(?csm_tip, "got new CSM state");
+
+    // Update the new state.
+    fcm_state.cur_csm_state = Arc::new(cs.clone());
+
+    if let Some(sync) = cs.sync() {
```
[the remark this answers is not in the document]
Comment (bewakes) [reply crates/consensus-logic/src/fork_choice_manager.rs]: Totally. this was missed.

**Comment by bewakes** [comment]
Actually there are other places too where the new Status interfaces will be used so will be done in the next PR, keeping this PR scoped.

**Review by delbonis — APPROVED** [review]
Alright this is good for now, there will probably be some more reworks in the future after the epoch rework happens but that's an issue for later.

**Comment by delbonis** [comment]
Force merged because of codecov.

**Pull request merged**